contract_api/application/service/registry_service.py

```diff
@@ -1,16 +1,85 @@
-from contract_api.domain.models.offchain_service_attribute import OffchainServiceAttribute
+import os
+import tempfile
+import uuid
+from datetime import datetime as dt
+from pathlib import Path
+
+from common.boto_utils import BotoUtils
+from common.constant import BuildStatus
+from common.utils import Utils, extract_zip_file, make_tarfile, download_file_from_url
+from contract_api.config import REGION_NAME, ASSETS_COMPONENT_BUCKET_NAME
+from contract_api.domain.factory.service_factory import ServiceFactory
+from contract_api.domain.models.demo_component import DemoComponent
 from contract_api.infrastructure.repositories.service_repository import OffchainServiceConfigRepository
 
+utils = Utils()
+boto_utils = BotoUtils(region_name=REGION_NAME)
 offchain_service_config_repo = OffchainServiceConfigRepository()
+service_factory = ServiceFactory()
 
 
 class RegistryService:
     def __init__(self, org_id, service_id):
         self.org_id = org_id
         self.service_id = service_id
 
-    def save_offchain_service_attribute(self, attributes):
-        offchain_service_attribute = OffchainServiceAttribute(org_id=self.org_id, service_id=self.service_id,
-                                                              attributes=attributes)
-        offchain_service_attribute = offchain_service_config_repo.save_offchain_service_attribute(offchain_service_attribute)
+
+
+    def publish_offline_assets(self, attributes):
+        # publish demo component file
+        if attributes.get("demo_component_url", ""):
+            new_demo_url = self.publish_demo_component(attributes["demo_component_url"])
+            attributes.update({"demo_component_status": "PENDING"})
+            attributes.update({"demo_component_url": new_demo_url})
+            attributes.update({"demo_component_last_updated": str(dt.utcnow())})
+        return attributes
+
+    def save_offchain_service_attribute(self, new_offchain_attributes):
+        existing_offchain_attributes = offchain_service_config_repo.get_offchain_service_config(self.org_id,
+                                                                                                self.service_id)
+        demo_component = self.create_or_update_demo_component_domain_model(new_offchain_attributes,
+                                                                           existing_offchain_attributes.to_dict()[
```

Comment:
existing_offchain_attributes.to_dict()["attributes"]
Let's make use of extra line.